### DISCUSSION TITLE
fieldType permissionResource lost in frontend-permission-toolkit

### DISCUSSION THREAD

**Discussion by langert** [pimcore/pimcore#15906] [Q&A / Support] [answered]
In Pimcore 11, I can add new fields to my DataObject Class using the permission Tookit. Nice. 

Now adding a fieldType "Permission Resource" > the key icon is shown and the field is generated. 
After saving the DataObject Class, close it and try to re-edit > the fieldTyp is no longer from Type "Permission Resource" - its just a simple "select" with the options inherit, allow, deny.

Same procedure with Type: Dynamic Permission Resource > fieldType persists after saving and reopen the DataObject Class.

In pimcore/var/classes/definition-MyClass.php

`
 * - test [select]
 ...
5 => \FrontendPermissionToolkitBundle\CoreExtensions\ClassDefinitions\PermissionResource::__set_state(array(
                 'name' => 'test',
                 'title' => 'Test',
                 'tooltip' => '',
                 'mandatory' => false,
                 'noteditable' => false,
                 'index' => false,
                 'locked' => false,
                 'style' => '',
                 'permissions' => NULL,
                 'fieldtype' => 'permissionResource',
                 'relationType' => false,
                 'invisible' => false,
                 'visibleGridView' => false,
                 'visibleSearch' => false,
                 'blockedVarsForExport' => 
                array (
                ),
                 'options' => NULL,
                 'defaultValue' => NULL,
                 'optionsProviderClass' => NULL,
                 'optionsProviderData' => NULL,
                 'columnLength' => 190,
                 'dynamicOptions' => false,
                 'defaultValueGenerator' => '',
                 'width' => NULL,
              )),
`
Bug, configuration error or is the excpected behavior?

**Comment by blankse** [comment, marked as answer]
@langert It's a bug of frontend-permission-toolkit. I created a PR https://github.com/pimcore/frontend-permission-toolkit/pull/58

**Reply by langert** [reply]
thx @blankse, then I will hotfix it, following your PR while waiting for an update
